### ISSUE TITLE
Notice of deprecation of Node.js 8 without semver major bump of Architect

### ISSUE THREAD

**Issue by ryanblock** [architect/architect#832]
This is just a notice to let folks know that we are technically implementing a "breaking" change in Architect by removing Node.js 8 from all of our testing; in the weeks to come, we will also begin swapping out one of the [only known Node.js 8-specific operations](https://github.com/architect/architect/issues/814) found throughout our codebase for a >= 10 equivalent builtin.

What's important to note, however, is that we're **not bumping the semver major** for Architect to `7`. Why, you may ask?

Architect deploys to Lambda, and Lambda went EOL on `nodejs8.10` 5 months ago – which means for the last 5 months, if you were deploying Node.js 8-based projects you were seeing failures from AWS. As such, we can probably safely assume that folks upgraded their Node.js environments both locally and in their app.

tl;dr we are downstream of AWS on runtime deprecations, so while it's a bit fuzzy of a decision, we are opting not to do a semver major bump for this (and likely any future Node.js runtime deprecation events in Lambdaland).
